cli/tests/test_git_fetch.rs

```diff
@@ -1755,3 +1779,47 @@ fn test_git_fetch_remote_only_bookmark(subprocess: bool) {
     "###);
     }
 }
+
+#[test_case(false; "use git2 for remote calls")]
+#[test_case(true; "spawn a git subprocess for remote calls")]
+fn test_git_fetch_multiple_remotes_same_commit(subprocess: bool) {
+    fn setup_env(subprocess: bool) -> (TestEnvironment, PathBuf) {
+        let test_env = TestEnvironment::default();
+        if subprocess {
+            test_env.set_up_git_subprocessing();
+        }
+        test_env.add_config("git.auto-local-bookmark = true");
+        test_env.jj_cmd_ok(test_env.env_root(), &["git", "init", "repo"]);
+        let repo_path = test_env.env_root().join("repo");
+        add_git_remote(&test_env, &repo_path, "rem1");
+        clone_git_add_commit(&test_env, "rem1", "rem2");
+        test_env.jj_cmd_ok(&repo_path, &["git", "remote", "add", "rem2", "../rem2"]);
+
+        (test_env, repo_path.clone())
+    }
+
+    // fetch with different orderings
+    let (test_env1, repo_path1) = setup_env(subprocess);
+    test_env1.jj_cmd_ok(
+        &repo_path1,
+        &["git", "fetch", "--remote", "rem1", "--remote", "rem2"],
+    );
+    let (test_env2, repo_path2) = setup_env(subprocess);
+    test_env2.jj_cmd_ok(
+        &repo_path2,
+        &["git", "fetch", "--remote", "rem2", "--remote", "rem1"],
+    );
+
+    let bookmark_output1 = get_bookmark_output(&test_env1, &repo_path1);
+    let bookmark_output2 = get_bookmark_output(&test_env2, &repo_path2);
+    assert_eq!(bookmark_output1, bookmark_output2);
```

Comment:
It's not important that the conflicts are sorted by remote.

A bigger problem reported in #4920 is that an abandoned commit in one remote becomes hidden (or abandoned) even if the commit is referenced by the other remote, and the descendants were rebased iirc. Can you add the test for this scenario?